packages/valory/skills/decision_maker_abci/behaviours/sell_outcome_token.py

```diff
@@ -0,0 +1,181 @@
+            contract_address=self.market_maker_contract_address,
+            contract_id=str(FixedProductMarketMakerContract.contract_id),
+            contract_callable="get_sell_data",
+            return_amount=self.return_amount,
+            outcome_index=self.outcome_index,
+            max_outcome_tokens_to_sell=self.sell_amount,
+        )
+        if response_msg.performative != ContractApiMessage.Performative.STATE:
+            self.context.logger.error(
+                f"Could not get the data for the buy transaction: {response_msg}"
+            )
+            return False
+
+        sell_data = response_msg.state.body.get("data", None)
+        if sell_data is None:
+            self.context.logger.error(
+                f"Something went wrong while trying to encode the buy data: {response_msg}"
+            )
+            return False
+
+        batch = MultisendBatch(
+            to=self.market_maker_contract_address,
+            data=HexBytes(sell_data),
+        )
+        self.multisend_batches.append(batch)
+        return True
+
+    def _prepare_safe_tx(self) -> Generator[None, None, Optional[str]]:
+        """Prepare the safe transaction for selling an outcome token and return the hex for the tx settlement skill."""
+        for step in (
+            self._build_approval_tx,
+            self._calc_sell_amount,
+            self._build_sell_tx,
+            self._build_multisend_data,
+            self._build_multisend_safe_tx_hash,
+        ):
+            yield from self.wait_for_condition_with_sleep(step)
+
+        outcome = self.sampled_bet.get_outcome(self.outcome_index)
+        investment = self._collateral_amount_info(self.return_amount)
+        self.context.logger.info(
+            f"Preparing a multisig transaction to sell the outcome token for {outcome!r}, with confidence "
+            f"{self.synchronized_data.confidence!r}, for the amount of {investment}, which is equal to the amount of "
+            f"{self.sell_amount!r} WEI of the conditional token corresponding to {outcome!r}."
+        )
+
+        return self.tx_hex
+
+    def async_act(self) -> Generator:
+        """Do the action."""
+
+        agent = self.context.agent_address
+
+        with self.context.benchmark_tool.measure(self.behaviour_id).local():
+            tx_submitter = betting_tx_hex = mocking_mode = None
+
+            # if the vote is the same as the previous vote then there is no change in the supported outcome, so we
+            # should not sell
+            if self.synchronized_data.vote == self.synchronized_data.previous_vote:
+                payload = MultisigTxPayload(
+                    agent, tx_submitter, betting_tx_hex, mocking_mode
+                )
+
+                yield from self.finish_behaviour(payload)
+
+            self.return_amount = self.bets[
+                self.synchronized_data.sampled_bet_index
+            ].invested_amount
```

Comment:
So why isn't the `return_amount` a method or property since it is only set here?